### PR TITLE
Update online ASR tutorial

### PR DESCRIPTION
**Pull request by mthrok**
https://554729-90321822-gh.circle-artifacts.com/0/docs/tutorials/online_asr_tutorial.html

1. Add figure to explain the caching
2. Fix the initialization of stream iterator

### CODE REVIEW

File: examples/tutorials/online_asr_tutorial.py [hunk 1]
```diff
@@ -72,7 +72,11 @@
 
 ######################################################################
 # Streaming inference works on input data with overlap.
-# Emformer RNN-T expects right context like the following.
+# Emformer RNN-T model treats the newer portion of the input data,
+# denoted as "right context", a preview of future context.
```
Comment (hwangjeff) [inline examples/tutorials/online_asr_tutorial.py:76]:
```suggestion
# Emformer RNN-T model treats the newest portion of the input data
# as the "right context" — a preview of future context.
```

File: examples/tutorials/online_asr_tutorial.py [hunk 2]
```diff
@@ -72,7 +72,11 @@
...
+# In the subsequent inference call, the model expects the main segment
```
Comment (hwangjeff) [inline examples/tutorials/online_asr_tutorial.py:77]:
```suggestion
# In each inference call, the model expects the main segment
```

File: examples/tutorials/online_asr_tutorial.py [hunk 3]
```diff
@@ -115,14 +119,20 @@
 print(streamer.get_out_stream_info(0))
 
 ######################################################################
-# `Streamer` iterate the source media without overlap, so we make a
-# helper structure that caches a chunk and return it with right context
-# appended when the next chunk is given.
+# As previously explained, Emformer RNN-T model expects input data with
+# overlaps, however, `Streamer` iterates the source media without overlap,
```
Comment (hwangjeff) [inline examples/tutorials/online_asr_tutorial.py:123]:
```suggestion
# overlaps; however, `Streamer` iterates the source media without overlap,
```

File: examples/tutorials/online_asr_tutorial.py [hunk 4]
```diff
@@ -115,14 +119,20 @@
...
+# so we make a helper structure that caches a part of input data from
+# `Streamer` as right context, then append it to the next input data from
```
Comment (hwangjeff) [inline examples/tutorials/online_asr_tutorial.py:125]:
```suggestion
# `Streamer` as right context and then appends it to the next input data from
```

**Review by hwangjeff — APPROVED** [review]
just some nits — looks good

side thought: do you think it'd be helpful to be explicit about how the quality of the transcriptions is as expected partly because the model is trained on librispeech but is used to perform inference on non-librispeech data?

**Comment by mthrok** [comment]
> side thought: do you think it'd be helpful to be explicit about how the quality of the transcriptions is as expected partly because the model is trained on librispeech but is used to perform inference on non-librispeech data?

How would you phrase it?

I gave some thoughts at the very beginning, but if we mention that the sample is out-of-domain data, then the narrative is "it's not that good and the reason is the sample is out-of-domain". So I decided not to mention.

**Pull request closed without merging**